packages/ods-react/src/components/combobox/src/components/combobox/Combobox.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Combobox as VendorCombobox,
+  createListCollection,
+} from '@ark-ui/react/combobox';
+import { type FC, type JSX, forwardRef, useEffect, useState } from 'react';
+import { type ComboboxInputValueChangeDetails, type ComboboxProp, ComboboxProvider, type ComboboxValueChangeDetails } from '../../context/useCombobox';
+import { findLabelForValue, flattenItems, getFilteredItems } from '../../controller/combobox';
+
+const Combobox: FC<ComboboxProp> = forwardRef(({
+  allowCustomValue = true,
+  children,
+  className,
+  customOptionRenderer,
+  defaultValue,
+  disabled,
+  highlightResults = false,
+  items,
+  newElementLabel = 'Add ',
+  noResultLabel = 'No results found',
+  onValueChange,
+  readOnly,
+  value,
+  ...props
+}, ref): JSX.Element => {
+  const [inputValue, setInputValue] = useState('');
+  const [userIsTyping, setUserIsTyping] = useState(false);
+  const currentValue = value && value.length > 0 ? value : defaultValue;
+
+  /**
+   * userIsTyping prevents input reset during search while allowing
+   * synchronization with selected values when not typing
+   */
+  useEffect(() => {
+    if (currentValue && currentValue.length > 0 && !userIsTyping) {
+      const label = findLabelForValue(items, currentValue[0]);
+      setInputValue(label);
+    }
+  }, [currentValue, items, userIsTyping]);
+
+  const filteredItems = getFilteredItems({
+    allowCustomValue,
+    customOptionRenderer,
+    inputValue,
+    items,
+    newElementLabel,
+    value: currentValue,
+  });
+
+  const flatItems = flattenItems(filteredItems);
+  const collection = createListCollection({ items: flatItems });
```

Comment:
`collection`, `filteredItems` and `flatItems` are recalculted on every render, you should use `useMemo` to memoize those values so that they're recomputed only when a used value change.